logic/src/commonMain/kotlin/com/wire/kalium/logic/data/connection/ConnectionRepository.kt

```diff
@@ -202,13 +205,43 @@ internal class ConnectionDataSource(
                             isService = userProfileDTO.service != null
                         )
                     )
+                    insertConversationFromConnection(connection)
                     userDAO.insertUser(userEntity)
-                    // todo: ask if we need to insert the conversation here when the connection is sent!
                     connectionDAO.insertConnection(connectionMapper.modelToDao(connection))
                 }
             })
         }
 
+    private suspend fun insertConversationFromConnection(connection: Connection) {
+        when (connection.status) {
+            SENT -> conversationRepository.fetchConversation(connection.qualifiedConversationId)
+            PENDING -> {
+                /* TODO: we had to do it manually, the server won't give us for received connections
+                     as the final solution we need to ignore the conversation part, but now? we can't! */
+                conversationDAO.insertConversation(
+                    conversationEntity = ConversationEntity(
+                        id = idMapper.toDaoModel(connection.qualifiedConversationId),
+                        name = null,
+                        type = ConversationEntity.Type.CONNECTION_PENDING,
+                        teamId = null,
+                        protocolInfo = ConversationEntity.ProtocolInfo.Proteus,
+                        creatorId = connection.from,
+                        lastNotificationDate = null,
+                        lastModifiedDate = connection.lastUpdate,
+                        lastReadDate = connection.lastUpdate,
+                        access = emptyList(),
+                        accessRole = emptyList(),
+                        isCreator = false
+                    )
+                )
+            }
+
+            else -> {
+                kaliumLogger.i("INSERT CONVERSATION FROM CONNECTION NOT ENGAGED FOR $connection")
+            }
```

Comment:
Just one up on @yamilmedina's suggestion: No `else`, but rather list all possible values. This way if we add a new value we will have a compilation error instead of a possible silent failure.

```suggestion
            NOT_CONNECTED, BLOCKED, IGNORED, CANCELLED,
            MISSING_LEGALHOLD_CONSENT, ACCEPTED -> {
                kaliumLogger.i("INSERT CONVERSATION FROM CONNECTION NOT ENGAGED FOR $connection")
            }
```